cql3/statements/create_index_statement.cc

```diff
@@ -67,13 +68,13 @@ create_index_statement::validate(query_processor& qp, const service::client_stat
     if (_raw_targets.empty() && !_properties->is_custom) {
         throw exceptions::invalid_request_exception("Only CUSTOM indexes can be created without specifying a target column");
     }
-
+    
     _properties->validate();
 }
 
 std::vector<::shared_ptr<index_target>> create_index_statement::validate_while_executing(data_dictionary::database db) const {
     auto schema = validation::validate_column_family(db, keyspace(), column_family());
-
+    auto classes = db.find_column_family(schema).get_index_manager().supported_custom_classes(db.features());
```

Comment:
Seems like the supported classes shouldn't depend on the current table?

Reply:
Moreover, as far as I can tell, you don't even use the list of custom classes - you just want to check if a certain custom class is allowed. So it could have been a static function (in the index_manager class) called is_custom_class_supported(name). I'm not sure why db.features() is needed here, but I guess that can be a parameter too...